ID.me WebVerify SDK/IDmeWebVerify.m

```diff
@@ -95,91 +97,126 @@ - (void)verifyUserInViewController:(UIViewController *)externalViewController
+                                                    //TODO: refresh token
+                                                }
+                                            } else {
+                                                NSError *modifiedError = [[NSError alloc] initWithDomain:IDME_WEB_VERIFY_ERROR_DOMAIN
+                                                                                                    code:IDmeWebVerifyErrorCodeVerificationDidFailToFetchUserProfile
+                                                                                                userInfo:error.userInfo];
+                                                webVerificationResults(nil, modifiedError, nil);
+                                            }
+                                        } else {
+                                            NSError *modifiedError = [[NSError alloc] initWithDomain:IDME_WEB_VERIFY_ERROR_DOMAIN
+                                                                                                code:IDmeWebVerifyErrorCodeVerificationDidFailToFetchUserProfile
+                                                                                            userInfo:error.userInfo];
+                                            webVerificationResults(nil, modifiedError, nil);
+                                        }
+
+                                        // Dismiss _webViewController and clear _webView cache
+                                        if (self.webNavigationController) {
+                                            [self destroyWebNavigationController:self];
+                                        }
+                                    });
+                                    
+                                }];
+                                [task resume];
     }];
 }
 
 -(void)logout {
     [self.keychainData clean];
 }
 
-- (void)getAccessTokenWithScope:(NSString* _Nonnull)scope forceRefreshing:(BOOL)force result:(IDmeVerifyWebVerifyResults _Nonnull)callback;{
+- (void)getAccessTokenWithScope:(NSString* _Nullable)scope forceRefreshing:(BOOL)force result:(IDmeVerifyWebVerifyResults _Nonnull)callback{
+
     if (force) {
         // TODO: refresh token
         callback(nil, nil, nil);
         return;
     }
 
-    if (![scope isEqualToString:self.keychainData.scope]) {
-        [self logout];
-        [self verifyUserInViewController:[self topMostController] scope:scope withTokenResult:callback];
+    NSString* token;
+    NSDate* expiration;
+    if (scope) {
+        // get the token for the specified scope
+        token = [self.keychainData accessTokenForScope:scope];
+        expiration = [self.keychainData expirationDateForScope:scope];
+
+    } else {
+        // get last used token
+        NSString* latestScope = [self.keychainData getLatestUsedScope];
+
+        if (!latestScope) {
+            // no token has been requested yet
+            callback(nil, nil, nil);
```

Comment:
This happens when there has never been a login. We could return an error `IDmeWebVerifyErrorCodeNoSuchScope`